playground/backend/internal/components/cache_component.go

```diff
@@ -37,73 +38,129 @@ func NewService(cache cache.Cache, db db.Database) *CacheComponent {
 
 // GetSdkCatalogFromCacheOrDatastore returns the sdk catalog from the cache
 // - If there is no sdk catalog in the cache, gets the sdk catalog from the Cloud Datastore and saves it to the cache
-func (cp *CacheComponent) GetSdkCatalogFromCacheOrDatastore(ctx context.Context) ([]*entity.SDKEntity, error) {
-	sdks, err := cp.cache.GetSdkCatalog(ctx)
-	if err != nil {
-		logger.Errorf("error during getting the sdk catalog from the cache, err: %s", err.Error())
-		sdks, err = cp.db.GetSDKs(ctx)
+func (cp *CacheComponent) GetSdkCatalogFromCacheOrDatastore(ctx context.Context, cacheRequestTimeout time.Duration) ([]*entity.SDKEntity, error) {
+	sdksCh := make(chan []*entity.SDKEntity, 1)
+	errCh := make(chan error, 1)
+	go func() {
+		sdks, err := cp.cache.GetSdkCatalog(ctx)
```

Comment:
let's just do cache request with a reduced timeout:
```
ctx = context.WithTimeout(ctx, CACHE_TO)
```

no need for goroutines here